### ISSUE TITLE
bug(mat-chip-listbox): fails on accessibility score

### ISSUE THREAD

**Issue by aalbericio** [angular/components#31222]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Whenever I use "mat-chip-listbox", I can't make it score 100% on Lighthouse. It always complains about roles of children, etc.

The same code can be changed to use a "mat-chip-grid" and everything works as expected (scores 100%), so there must be something in this component (or subcomponents like mat-chip-option) that is not really working for accessibility.

### Reproduction

**StackBlitz** link: https://stackblitz.com/edit/6mumju4m-rs7e6l4g?file=src%2Fexample%2Fchips-autocomplete-example.html

Steps to reproduce:
1. Run the application in external mode and run a Lighthose accessibility-only report.
2. Get 21/22 points and some complaints about accessibility

### Expected Behavior

I would expect to work as mat-chip-grid (with mat-chip-rows) that score a 100%.

### Actual Behavior

Score is 21/22 with some complaints about :

Elements with an ARIA [role] that require children to contain a specific [role] are missing some or all of those required children.

![Image](https://github.com/user-attachments/assets/541caa5c-4e7a-429f-b9a7-ad041505baea)

### Environment

- Angular: 19.2.0
- CDK/Material: 19.2.17 (same example as Angular Material official website)
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
This is because you're using `matChipRemove` with the listbox, but it's actually meant for the chip grid. A remove button doesn't make sense inside a selectable chip.
